packages/apollo-collaboration-server/src/features/features.service.ts

```diff
@@ -0,0 +1,426 @@
+    // Search correct feature
+    const features = await this.featureModel
+      .find({
+        'gff3FeatureLineWithRefs.start': { $lte: searchDto.end },
+        'gff3FeatureLineWithRefs.end': { $gte: searchDto.start },
+        'gff3FeatureLineWithRefs.seq_id': searchDto.seq_id,
+      })
+      .exec()
+    this.logger.debug(
+      `Searching features for Seq_id=${searchDto.seq_id}=, Start=${searchDto.start}=, End=${searchDto.end}=`,
+    )
+
+    if (!features) {
+      const errMsg = `ERROR: No features were found in database`
+      this.logger.error(errMsg)
+      throw new NotFoundException(errMsg)
+    }
+
+    this.logger.debug(
+      `The following feature(s) matched  = ${JSON.stringify(features)}`,
+    )
+    return features
+  }
+
+  /**
+   * Get feature by featureId. When retrieving features by id, the features and any of its children are returned, but not any of its parent or sibling features.
+   * @param featureid - featureId
+   * @returns Return the feature(s) if search was successful. Otherwise throw exception
+   */
+  async getFeatureByFeatureId(featureId: string) {
+    // Search correct feature
+    const featureObject = await this.featureModel.findOne({ featureId }).exec()
+
+    if (!featureObject) {
+      const errMsg = `ERROR: The following featureId was not found in database ='${featureId}'`
+      this.logger.error(errMsg)
+      throw new NotFoundException(errMsg)
+    }
+
+    const updatableObjectAsGFFItemArray =
+      featureObject.gff3FeatureLineWithRefs as unknown as GFF3FeatureLineWithRefs[]
+    this.logger.verbose(`Feature found  = ${JSON.stringify(featureObject)}`)
+    // Now we need to find correct top level feature or sub-feature inside the feature
+    const foundFeature = await this.getObjectByFeatureId(
+      updatableObjectAsGFFItemArray,
+      featureId,
+    )
+    if (!foundFeature) {
+      const errMsg = `ERROR when searching feature by featureId`
+      this.logger.error(errMsg)
+      throw new NotFoundException(errMsg)
+    }
+    this.logger.debug(`Feature found: ${JSON.stringify(foundFeature)}`)
+    return foundFeature
+  }
+
+  /**
+   * Get single feature by featureId
+   * @param featureObject -
+   * @param featureId -
+   * @returns
+   */
+  async getObjectByFeatureId(
+    featureObject: GFF3FeatureLineWithRefs[],
+    featureId: string,
+  ) {
+    // Loop all lines and add those into cache
+    for (const entry of featureObject) {
+      this.logger.verbose(`Entry=${JSON.stringify(entry)}`)
+      if (entry.hasOwnProperty('featureId')) {
+        const assignedVal: GFF3FeatureLineWithRefsAndFeatureId =
+          Object.assign(entry)
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.logger.verbose(`Top level featureId=${assignedVal.featureId!}`)
+        // If matches
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        if (assignedVal.featureId! === featureId) {
```

Comment:
I don't think these assertions and eslint-disables are necessary

Reply:
Without _assertion_ and _eslint-disables_, I'm getting the following warnings:
  _297:52  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
  299:13  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion_